spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -73,7 +73,12 @@ public boolean caseSensitive() {
 
   public boolean localityEnabled() {
     boolean defaultValue = Util.mayHaveBlockLocations(table.io(), table.location());
-    return PropertyUtil.propertyAsBoolean(readOptions, SparkReadOptions.LOCALITY, defaultValue);
+    return confParser
+        .booleanConf()
+        .option(SparkReadOptions.LOCALITY)
+        .sessionConf(SparkSQLProperties.LOCALITY_ENABLED)
+        .defaultValue(defaultValue)
+        .parse();
```

Comment:
On further thought, I think it would be better to do
```
    if (defaultValue) {
      return confParser
          .booleanConf()
          .option(SparkReadOptions.LOCALITY)
          .sessionConf(SparkSQLProperties.LOCALITY_ENABLED)
          .defaultValue(true)
          .parse();
    } else {
      return false;
    }
```
If defaultValue is false, then locality should be disabled, regardless of the option or session conf, as block locations will not be available.

Reply:
Is there any issue with passing through "true" when that has no effect?

Reply:
I would probably rename "defaultValue" to "hasBlockLocations" or "canReadLocal" or something

Reply:
If `SparkReadConf#localityEnabled` returns true, even though the table is not in HDFS, then Iceberg will try to get the block locations, and I am not sure about this, since I haven't tested it, I think the HDFS call could throw an IOException and Iceberg will rethrow that. That would be a bad thing.

Reply:
Actually, maybe it wouldn't throw an exception. But the default implementation of `FileSystem#getFileBlockLocations` does
```
                String[] name = new String[]{"localhost:50010"};
                String[] host = new String[]{"localhost"};
                return new BlockLocation[]{new BlockLocation(name, host, 0L, file.getLen())};
```
and from that, we get a `String[]` of the hosts, and that probably won't be too helpful to Spark.

Reply:
+1 to `hasBlockLocations`.